server/qrops.go

```
@@ -0,0 +1,291 @@
+	}
+
+	decodedURL, err := base64.StdEncoding.DecodeString(qrURL[0])
+
+	if err != nil {
+		return err
+	}
+
+	q.DecodedQRURL = string(decodedURL)
+	return nil
+}
+
+func (q *QRConfig) setAllowProfileImage() {
+	allowProfileImage, ok := q.Params["allowProfileImage"]
+
+	if !ok || len(allowProfileImage) == 0 {
+		// we default to false when this flag was not provided
+		// so someone does not want to allowProfileImage on their QR Image
+		// fine then :)
+		q.WithLogo = false
+	}
+
+	LogoOnImage, err := strconv.ParseBool(allowProfileImage[0])
+
+	if err != nil {
+		// maybe for fun someone tries to send non-boolean values to this flag
+		// we also default to false in that case
+		q.WithLogo = false
+	}
+
+	// if we reach here its most probably true
+	q.WithLogo = LogoOnImage
+}
+
+func (q *QRConfig) setErrorCorrectionLevel() {
+	level, ok := q.Params["level"]
+	if !ok || len(level) == 0 {
+		// we default to MediumLevel of error correction when the level flag
+		// is not passed.
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionMedium)
+	}
+
+	levelInt, err := strconv.Atoi(level[0])
+	if err != nil || levelInt < 0 {
+		// if there is any issue with string to int conversion
+		// we still default to MediumLevel of error correction
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionMedium)
+	}
+
+	switch levelInt {
+	case 1:
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionLow)
+	case 2:
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionMedium)
+	case 3:
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionQuart)
+	case 4:
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionHighest)
+	}
+}
+
+func (q *QRConfig) setSize() error {
+	size, ok := q.Params["size"]
+
+	if ok {
+		imageToBeResized, err := strconv.Atoi(size[0])
+
+		if err != nil || imageToBeResized <= 0 {
+			return err
+		}
```

If the err is nil, but `imageToBeResized` is 0 or less, this would return a nil error. But you should return a specific error should be returned for this kind of issue.

---

fixed here : https://github.com/status-im/status-go/pull/3172/commits/8c97f5aeffb5d9804cfa144fbb3f6c315c1ca32d